modules/express/src/expressApp.ts

```diff
@@ -150,10 +149,13 @@ function configureProxy(app: express.Application, config: Config): void {
  * @param app
  * @return {Server}
  */
-async function createHttpsServer(app: express.Application, config: Config): Promise<https.Server> {
+async function createHttpsServer(
+  app: express.Application,
+  config: Config & { keyPath: string; crtPath: string }
```

Comment:
What does the `&` and the object afterwards do? Is that a way to immediately have access to properties of the config object? In that case, wouldn't you no more need the `const { keyPath, crtPath } = config;` line anymore?

Reply:
This is an [intersection type](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#intersection-types) which defines the type of the `config` parameter. It's saying that the type is `Config` (which has optional `keyPath` and `crtPath` properties) and'ed with the second type, which has non-optional `keyPath` and `crtPath` properties. This allows us to use these properties without checking them first in `createHttpsServer`.

We already check these properties in `isTLS` so there's no need to check them again here.

Reply:
This change was required since I removed the `Bluebird.promisifyAll` and just used the new `fs.promises` equivalents which are available starting in node 10. `Bluebird.promisifyAll` throws away all types and replaces them with `any` so we weren't able to do this at the type level before.